drivers/overlay/overlay.go

```diff
@@ -571,15 +573,17 @@ func (d *Driver) create(id, parent string, opts *graphdriver.CreateOpts) (retErr
 	if err := idtools.MkdirAllAs(path.Dir(dir), 0700, rootUID, rootGID); err != nil {
 		return err
 	}
+	perms := defaultPerms
 	if parent != "" {
 		st, err := system.Stat(d.dir(parent))
 		if err != nil {
 			return err
 		}
+		perms = os.FileMode(st.Mode())
```

Comment:
In all instances: AFAICS `st.Mode()` is already an `os.FileMode`, so the cast is unnecessary; OTOH the value also includes the file type bits.

So AFAICS this should be `perms = st.Mode().Perms()`.

Reply:
if I pass st.Mode() I get this error.

```
make
GO111MODULE=on go build -mod=vendor -compiler gc -tags "  "  ./cmd/containers-storage
# github.com/containers/storage/drivers/overlay
drivers/overlay/overlay.go:582:9: cannot use st.Mode() (type uint32) as type os.FileMode in assignment
make: *** [Makefile:53: containers-storage] Error 2
```
Which indicates that st.Mode() has no value of perms and needs the cast.

Note this is a system.Stat not an os.Stat.

Reply:
Ah, my mistake.